### ISSUE TITLE
Access to schema data

### ISSUE THREAD

**Issue by norpan** [NoRedInk/json-elm-schema#37]
I'm in the process of making an Elm substitute for https://github.com/mozilla-services/react-jsonschema-form which we are right now embedding, to get more control over how the form looks. The schema data type in NoRedInk/json-elm-schema is good enough for our purposes, but the small issue is that it's an opaque type.

Now, we _could_ just copy all the files and thereby get access to the actual `Model` but maybe there is some good way that we can have a better accessor API? Have you given this any thought? Or perhaps not much is lost if the type was exposed as-is?

**Comment by jwoudenberg** [comment]
Hi Norpan, that's so cool! I'm looking forward to seeing that in action!

The primary concern with exposing the type as-is, is that every change to the type (for instance: support some new schema-spec functionality) would result in a major version bump of the package.

One possibility is to put the type into a separate package, say `elm-json-schema-types`, and then have this library take that package as a dependency. Your library would also depend on `elm-json-schema-types`. `elm-json-schema-types` will have the major version bumps, but that's okay since almost no-one depends on it. The `json-elm-schema` package could keep its more stable API. Both packages could even live together in this repo.

Does that sound a good way forward to you? Copying over the files containing the types might still be a good way to start your form library though. Once you have something running we can evaluate whether it still seems a good idea and act accordingly.

Thoughts?

**Comment by norpan** [comment]
Yes, we'll probably start by copying (or cloning it and exposing the type). We are already using elm-github-install so we don't need to publish it.

I was just wondering if you had thought about an API for accessing the type, but perhaps it's best just to work with the raw data type, since you'd need to case match it anyway.
